x/subscription/keeper/subscription.go

```diff
@@ -0,0 +1,198 @@
+package keeper
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/lavanet/lava/utils"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	packages "github.com/lavanet/lava/x/packages/keeper"
+	"github.com/lavanet/lava/x/subscription/types"
+)
+
+// SetSubscription sets a subscription (of a consumer) in the store
+func (k Keeper) SetSubscription(ctx sdk.Context, sub types.Subscription) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.SubscriptionKeyPrefix))
+	b := k.cdc.MustMarshal(&sub)
+	store.Set(types.SubscriptionKey(sub.Consumer), b)
+}
+
+// GetSubscription returns the subscription of a given consumer
+func (k Keeper) GetSubscription(ctx sdk.Context, consumer string) (val types.Subscription, found bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefix(types.SubscriptionKeyPrefix))
+
+	b := store.Get(types.SubscriptionKey(consumer))
+	if b == nil {
+		return val, false
+	}
+
+	k.cdc.MustUnmarshal(b, &val)
+	return val, true
+}
+
+// RemoveSubscription removes the subscription (of a consumer) from the store
+func (k Keeper) RemoveSubscription(ctx sdk.Context, consumer string) {
+	sub, _ := k.GetSubscription(ctx, consumer)
+	packages.PutPackage(ctx, sub.PackageIndex, sub.Block)
```

Comment:
In the latest "plans" implementation, we didn't include a plan wrapper for the PutEntry function (like you wrote). You obviously couldn't have known this, but commenting so you can fix after "plans" is merged.
Do you think we should implement plan wrappers to the common library's API?

Reply:
also, deal with error? what if a package (a.k.a. plan) with (sub.PackageIndex, sub.Block) doesn't exist?

Reply:
Yes, please add wrappers - see my review on the Plan PR.
An error here is a bad bug - we took a reference, we should always be able to drop it; And anyway, there is nothing we can do if that happens: panic or ignore. 
Likewise, PutPlan() should not return an error. If anything it - or the underlying FixateStore - should generate a log message error.